File: autos-stock-client/src/app/services/modele.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment';
import { Observable } from 'rxjs';
import {Modele} from "../models/modele";

@Injectable({ providedIn: 'root' })
export class ModeleService {
  private base = `${environment.apiUrl}/modeles`;
  constructor(private http: HttpClient) {}
  listByMarque(idMarque: number): Observable<Modele[]> {
    return this.http.get<Modele[]>(`${this.base}?idMarque=${idMarque}`);
  }
  create(nom: string, idMarque: number): Observable<Modele> {
    return this.http.post<Modele>(this.base, { nom, idMarque });
  }
}

File: Autos-stock-manager-services/src/main/java/org/autostock/services/ModeleService.java
package org.autostock.services;

import org.autostock.dtos.ModeleCreateDto;
import org.autostock.dtos.ModeleDto;
import org.autostock.models.Modele;

import java.util.List;

public interface ModeleService extends IService<Modele, Long> {
    List<ModeleDto> listByMarque(Long idMarque);
    ModeleDto create(ModeleCreateDto dto);
}
